cmd/containerd-shim-runc-v2/task/service.go

```diff
@@ -723,17 +679,100 @@ func (s *service) send(evt interface{}) {
 	s.events <- evt
 }
 
-func (s *service) handleProcessExit(e runcC.Exit, c *runc.Container, p process.Process) {
-	if ip, ok := p.(*process.Init); ok {
-		// Ensure all children are killed
-		if runc.ShouldKillAllOnExit(s.context, c.Bundle) {
-			if err := ip.KillAll(s.context); err != nil {
-				log.G(s.context).WithError(err).WithField("id", ip.ID()).
-					Error("failed to kill init's children")
+// handleInitExit processes container init process exits.
+// This is handled separately from non-init exits, because there
+// are some extra invariants we want to ensure in this case, namely:
+// - for a given container, the init process exit MUST be the last exit published
+// This is achieved by:
+// - killing all running container processes
```

Comment:
Do we _have_ to kill the execs?  I understand that this is to bring behavior to be the same between the private pidns and shared pidns cases, but it is a behavior change and I'm not sure if we'd want that brought back to 1.7 or 1.6 (and the bug about losing exits definitely exists in both of those releases).

Reply:
I think it's fine to only kill in the shared-pid namespace case – if it's a private namespace, all the exec exits will have been reaped regardless when we receive the init exit.